### PR TITLE
PRIME-1648 Maintenance View Org Agreements

### PR DESCRIPTION
**Pull request by james-hollinger**
Adds org agreements to the site maintenance page.
Fair amount of refactoring on the back end, mainly breaking up the agreement service and creating the org and enrollee agreement services.

### CODE REVIEW

File: prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  selector: 'app-organization-toa-maintenance-list-page',
+  templateUrl: './organization-toa-maintenance-list-page.component.html',
+  styleUrls: ['./organization-toa-maintenance-list-page.component.scss'],
+  providers: [FormatDatePipe]
+})
+export class OrganizationToaMaintenanceListPageComponent implements OnInit {
+  public busy: Subscription;
+  public orgAgreementVersions: AgreementVersion[];
+  public AgreementTypeNameMap = AgreementTypeNameMap;
+  public previewingToa: AgreementVersion;
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.ts:22]: Referenced above can be removed.

File: prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+@Component({
+  selector: 'app-organization-toa-maintenance-list-page',
+  templateUrl: './organization-toa-maintenance-list-page.component.html',
+  styleUrls: ['./organization-toa-maintenance-list-page.component.scss'],
+  providers: [FormatDatePipe]
+})
+export class OrganizationToaMaintenanceListPageComponent implements OnInit {
+  public busy: Subscription;
+  public orgAgreementVersions: AgreementVersion[];
+  public AgreementTypeNameMap = AgreementTypeNameMap;
```
Comment (hogmann) [inline prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.ts:21]: This doesn't appear to be used in controller or template.

File: prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
+<app-page mode="full"
+          [busy]="busy"
+          [showToolbar]="true"
+          (back)="onBack()">
+  <app-page-header>TOA Maintenance</app-page-header>
+  <app-page-subheader *ngIf="!previewingToa">
```
[the remark this answers is not in the document]
Comment (james-hollinger) [reply prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.html:6]: Can this be removed from the Enrollee TOA Maintenance page as well?

[the remark this answers is not in the document]
Comment (hogmann) [reply prime-angular-frontend/src/app/modules/adjudication/pages/organization-toa-maintenance-list-page/organization-toa-maintenance-list-page.component.html]: Yes fine to remove both places

**Review by hogmann — APPROVED** [review]
LGTM

**Review by mtpultz — APPROVED** [review]

**Pull request merged**